### PR TITLE
[dhctl] feat(dhctl-for-commander): add ability to pass initial state and on-phase callback for dhctl operations

### PR DESCRIPTION
**Pull request by distorhead**
Depends on (merge only after this PR): https://github.com/deckhouse/deckhouse/pull/5714

### CODE REVIEW

File: dhctl/pkg/operations/common.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package operations
+
+import (
+	"errors"
+	"github.com/deckhouse/deckhouse/dhctl/pkg/state"
```
Comment (name212) [inline dhctl/pkg/operations/common.go:5]: sort imports

File: dhctl/pkg/operations/common.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+type OnPhaseFunc func(completedPhase OperationPhase, completedPhaseState DhctlState, nextPhase OperationPhase, nextPhaseCritical bool) error
+
+type OperationPhase string
+
+const (
+	BaseInfraPhase             OperationPhase = "BaseInfra"
+	ExecuteBashibleBundlePhase OperationPhase = "ExecuteBashibleBundle"
+	InstallDeckhousePhase      OperationPhase = "InstallDeckhouse"
+	CreateResourcesPhase       OperationPhase = "CreateResources"
```
Comment (name212) [inline dhctl/pkg/operations/common.go:18]: I think we need one more phase - installAdditionalMastersAndStaticNodes

Comment (name212) [reply dhctl/pkg/operations/common.go:18]: After install deckhouse

**Review by name212 — CHANGES_REQUESTED** [review]

File: dhctl/pkg/operations/common.go (hunk 2 above)
Comment (distorhead) [reply dhctl/pkg/operations/common.go:18]: I would propose to add such phase in the separate PR on top of existing. BTW in the last PR there is small refactor and introduction of PhasedExecutionContext, which is related to the phases.

**Comment by distorhead** [comment]
Merging this PR into single aggregated MR [feat/dhctl-for-commander-1](https://github.com/deckhouse/deckhouse/tree/feat/dhctl-for-commander-1)

**Pull request merged**